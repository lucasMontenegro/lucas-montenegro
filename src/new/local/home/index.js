import React from "react"
import Typography from "@material-ui/core/Typography"
import createBaseClient from "new/local/paperbase/createBaseClient"
import useStyles from "new/local/home/useStyles"
import translations from "new/local/home/translations"
export default function home (appName, routing) {
  const BaseClient = createBaseClient({
    appName,
    clientName: `home`,
    routing,
  })
  function Home (props) {
    const classes = useStyles()
    return (
      <BaseClient
        {...props}
        subtitles={translations.subtitles}
        icons={translations.icons}
      >
        <div className={classes.root}>
          <Typography variant="h5" component="h1">
            {translations.messages[props.languageCode]}
          </Typography>
        </div>
      </BaseClient>
    )
  }
  Home.propTypes = createBaseClient.propTypes
  return Home
}