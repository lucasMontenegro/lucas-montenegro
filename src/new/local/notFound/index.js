import React from "react"
import createBaseClient from "new/local/paperbase/createBaseClient"
import NotFoundView from "new/local/notFound/NotFoundView"
import useReferrer from "new/local/notFound/useReferrer"
export default function notFound (appName, routing) {
  const BaseClient = createBaseClient({
    appName,
    clientName: `notFound`,
    routing,
  })
  function NotFound (props) {
    const { languageCode, match, location } = props
    const referrer = useReferrer(match, location)
    return (
      <NotFoundView
        languageCode={languageCode}
        match={match}
        referrer={referrer}
        routing={routing}
        BaseClient={BaseClient}
        clientProps={props}
      />
    )
  }
  NotFound.propTypes = createBaseClient.clientPropTypes
  return NotFound
}