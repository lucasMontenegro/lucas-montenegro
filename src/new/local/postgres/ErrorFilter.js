import monitor from "pg-monitor"
export default function ErrorFilter () {
  const matchers = []
  return {
    ignore (fn) {
      matchers.push(fn)
    },
    logger (err, e) {
      // usage:
      // pg.ignoreDbError(function matchUniqueUsernameError (err, e) {
      //   return (
      //     typeof err === `object`
      //     && err.code == `23505`
      //     && err.table == `usernames`
      //     && err.constraint == `usernames_pkey`
      //   )
      // })
      // pg.ignoreDbError === logger
      if (!matchers.find(fn => fn(err, e))) {
        monitor.error(err, e)
      }
    },
  }
}