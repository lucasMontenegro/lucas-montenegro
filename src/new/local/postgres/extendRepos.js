// Extend "db" object, i.e.:
// const { default: Repo } = require("new/local/Repo")
exports.default = function extendRepos (db, dc) {
  // cont.:
  // db.repo = new Repo(db, dc)
}