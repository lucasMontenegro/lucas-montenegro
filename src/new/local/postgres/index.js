import startDb from "new/local/postgres/startDb"
export const { db, pgp, ignoreError } = startDb()
export function closeDb () {
  db.$config.pgp.end()
}