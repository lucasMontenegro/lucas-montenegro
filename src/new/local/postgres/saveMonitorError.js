const os = require("os")
const fs = require("fs")
const path = require("path")
const { default: simpleErrorCallback } = require("new/local/utils/simpleErrorCallback")
const { default: globals } = require("new/local/utils/globals")
const logFile = path.resolve(globals.process.env.NODE_PATH, `../pg-errors.log`)
exports.default = function saveMonitorError (msg, info) {
  let logText = os.EOL + msg
  if (info.time) {
    logText = os.EOL + logText
  }
  fs.appendFile(logFile, logText, simpleErrorCallback)
}