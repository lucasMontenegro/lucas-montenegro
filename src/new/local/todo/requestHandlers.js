import minifySql from "new/local/utils/minifySql"
import syncRequestHandler from "new/local/utils/syncRequestHandler"
import { db, pgp } from "new/local/postgres"
import extractRowCount from "new/local/utils/postgres/extractRowCount"
const templates = {
  create: minifySql(`
    INSERT INTO todo (user_id, description, priority)
    VALUES ($1, $2, $3)
  `),
  read: minifySql(`
    SELECT todo_id, description, priority, status, created_on, updated_on
    FROM todo
    WHERE user_id = $1
  `),
  update: minifySql(`
    UPDATE todo
    SET priority=$3, status=$4, updated_on=NOW()
    WHERE user_id = $1 AND todo_id = $2
  `),
  delete: minifySql(`
    DELETE FROM todo
    WHERE user_id = $1 AND todo_id = $2
  `),
}
export default {
  post: syncRequestHandler(async (req, res) => {
    const { body } = req
    await db.none(pgp.as.format(templates.create, [req.user.sub, body.description, body.priority]))
    res.end()
  }),
  get: syncRequestHandler(async (req, res) => {
    res.json(db.any(pgp.as.format(templates.read, req.user.sub)))
  }),
  put: syncRequestHandler(async (req, res) => {
    const { body } = req
    const values = [req.user.sub, req.params.todo_id, body.priority, body.status]
    await db.none(pgp.as.format(templates.update, values))
    res.end()
  }),
  delete: syncRequestHandler(async (req, res) => {
    const { body } = req
    const values = [req.user.sub, req.params.todo_id]
    await db.result(pgp.as.format(templates.delete, values), null, extractRowCount)
    res.end()
  }),
}