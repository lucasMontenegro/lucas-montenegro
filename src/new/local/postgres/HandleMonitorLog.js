import saveMonitorError from "new/local/postgres/saveMonitorError"
import isProduction from "new/local/utils/isProduction"
export default function HandleMonitorLog () {
  const production = isProduction()
  return function handleMonitorLog (msg, info) {
    if (info.event === `error`) {
      production || saveMonitorError(msg, info)
    } else if (production) {
      info.display = false
    }
  }
}