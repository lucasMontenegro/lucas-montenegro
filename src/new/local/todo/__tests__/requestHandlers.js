import { db, pgp } from "new/local/postgres"
import requestHandlers from "new/local/todo/requestHandlers"
const mockFn = jest.fn()
const createMockFn = name => (...args) => {
  const arr = [name, args]
  mockFn(...arr)
  return arr
}
jest.mock(`new/local/postgres`, () => ({
  __esModule: true,
  db: {},
  pgp: { as: {} },
}))
pgp.as.format = (...args) => jestUtils.pgpInstance.as.format(...args)
;[`none`, `any`, `result`].reduce((db, name) => {
  db[name] = createMockFn(`db.${name}`)
  return db
}, db)
jest.mock(`new/local/utils/postgres/extractRowCount`, () => ({
  __esModule: true,
  default: `new/local/utils/postgres/extractRowCount`,
}))
const req = {
  user: { sub: `req.user.sub` },
  body: {
    description: `req.body.description`,
    priority: `req.body.priority`,
    status: `req.body.status`,
  },
  params: { todo_id: `req.params.todo_id` },
}
const res = [`end`, `json`].reduce((res, name) => {
  res[name] = createMockFn(`res.${name}`)
  return res
}, {})
describe(`local/todo/requestHandlers`, () => {
  afterEach(() => mockFn.mockClear())
  describe.each([[`post`], [`get`], [`put`], [`delete`]])(`requestHandlers.%s`, rest => {
    it(`should handle requests`, async () => {
      await requestHandlers[rest](req, res)
      expect(mockFn.mock.calls).toMatchSnapshot()
    })
  })
})