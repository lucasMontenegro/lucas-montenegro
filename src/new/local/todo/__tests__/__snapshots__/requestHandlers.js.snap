// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`local/todo/requestHandlers requestHandlers.delete should handle requests 1`] = `
Array [
  Array [
    "db.result",
    Array [
      "DELETE FROM todo WHERE user_id='req.user.sub' AND todo_id='req.params.todo_id'",
      null,
      "new/local/utils/postgres/extractRowCount",
    ],
  ],
  Array [
    "res.end",
    Array [],
  ],
]
`;

exports[`local/todo/requestHandlers requestHandlers.get should handle requests 1`] = `
Array [
  Array [
    "db.any",
    Array [
      "SELECT todo_id,description,priority,status,created_on,updated_on FROM todo WHERE user_id='req.user.sub'",
    ],
  ],
  Array [
    "res.json",
    Array [
      Array [
        "db.any",
        Array [
          "SELECT todo_id,description,priority,status,created_on,updated_on FROM todo WHERE user_id='req.user.sub'",
        ],
      ],
    ],
  ],
]
`;

exports[`local/todo/requestHandlers requestHandlers.post should handle requests 1`] = `
Array [
  Array [
    "db.none",
    Array [
      "INSERT INTO todo(user_id,description,priority)VALUES('req.user.sub','req.body.description','req.body.priority')",
    ],
  ],
  Array [
    "res.end",
    Array [],
  ],
]
`;

exports[`local/todo/requestHandlers requestHandlers.put should handle requests 1`] = `
Array [
  Array [
    "db.none",
    Array [
      "UPDATE todo SET priority='req.body.priority',status='req.body.status',updated_on=NOW()WHERE user_id='req.user.sub' AND todo_id='req.params.todo_id'",
    ],
  ],
  Array [
    "res.end",
    Array [],
  ],
]
`;
